app/jobs/experimentation/data_export_worker.rb

```diff
@@ -0,0 +1,64 @@
+module Experimentation
+  class DataExportWorker
+    require "csv"
+    include Sidekiq::Worker
+
+    attr_reader :experiment_name, :recipient_email_address, :results
+
+    def perform(experiment_name, recipient_email_address)
+      @experiment_name = experiment_name
+      @recipient_email_address = recipient_email_address
+      fetch_results
+      mail_csv
+    end
```

Comment:
Just to recap what we discussed on our call...I think it would be good if this call were just the 'traffic cop' here -- maybe something like this:

```
def perform
  @experiment = Experiment.find_by!(name: name)
  csv = Experimentation::Export.call(@experiment)
  Experimentation::Mailer.deliver_csv(csv, recipient)
end
```

The export doesn't need to know about the recipient or the mail, and the mail doesn't need to know about how the data is generated or queried.

Reply:
Oh, that would also let you write the CSV to disk on SBX pretty easily for inspection if emails weren't being sent from there.